### ISSUE TITLE
`cups-kyodialog3` fails to install with "do not know how to unpack source archive"

### ISSUE THREAD

**Issue by tim-hilt** [NixOS/nixpkgs#123694]
**Describe the bug**

`cups-kyodialog3` fails to install, because the package apparently doesn't know how to unpack the tar.gz-archive that contains the necessary `.ppd`s.

the install-output looks like this:

```
[sudo] password for tim:
building the system configuration...
error: builder for '/nix/store/lb728mjlxl38z6ahdlnrfb5fxlw4q8ih-source.drv' failed with exit code 1;
       last 10 log lines:
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       > 100   315  100   315    0     0    600      0 --:--:-- --:--:-- --:--:--   601
       > 100  107k    0  107k    0     0  82612      0 --:--:--  0:00:01 --:--:--  242k
       > unpacking source archive /build/Kyocera_Linux_PPD_Ver_8.1601.tar.gz
       >
       > gzip: stdin: not in gzip format
       > tar: Child returned status 1
       > tar: Error is not recoverable: exiting now
       > do not know how to unpack source archive /build/Kyocera_Linux_PPD_Ver_8.1601.tar.gz
       For full logs, run 'nix log /nix/store/lb728mjlxl38z6ahdlnrfb5fxlw4q8ih-source.drv'.
error: 1 dependencies of derivation '/nix/store/gd59h19bxjlvfzdlpf1qwjb52bg8m0kp-cups-kyodialog3-8.1601.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3ds8ik1s44l8bgwgyzgr4zg9gxz64ci3-cups-progs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/8kbwml6ana7sz62ldvc3dzs94mnd0z4n-pam-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/77qnsjdfra3n5s6xci5xfvy2ryh0iir4-set-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9mgy46hxf4hhklkii588w0khcqa4pykn-unit-script-cups-pre-start.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1rq07k34ql2nc8cbryh1s68kdz7qwsvg-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2mni8nzcnnqj0nwaxxha7n2sjgxzk6kp-nixos-system-x220-21.05.20210515.83d907f.drv' failed to build
error: getting status of '/tmp/nixos-rebuild.c2WPFg/result': No such file or directory
```

**To Reproduce**

1. Add `pkgs.cups-kyodialog3` to `service.printing.drivers`-list
2. `sudo nixos-rebuild switch`

**Expected behavior**

The `kyodialog3` printer-driver should be installed and working

**Notify maintainers**

@steveeJ 

**Metadata**

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.21, NixOS, 21.05.20210515.83d907f (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210503_6d2553a`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
 ```

**Comment by Artturin** [comment]
cc @mschuwalow you fixed it once before in #105445

**Comment by hexchen** [comment]
I have figured out the new download URL, which fixes this package. See https://github.com/NixOS/nixpkgs/pull/170820
